### PR TITLE
improved expand docstring

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

The `QuantumScript.expand` / `pennylane.tape.tape.expand_tape` can be a little confusing and is poorly document. It has several quirks that are only developer knowledge, and not explicitly written down.

**Description of the Change:**

Improvest the docstring for these methods.

**Benefits:**

Less confusion.

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-64625]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -29,6 +29,8 @@
 
 <h3>Documentation 📝</h3>
 
+* Improves the docstring for `QuantumScript.expand` and `qml.tape.tape.expand_tape`.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:32]:
```suggestion
* Improves the docstring for `QuantumScript.expand` and `qml.tape.tape.expand_tape`.
  [(#5974)](https://github.com/PennyLaneAI/pennylane/pull/5974)
```

**Review by PietropaoloFrisoni — COMMENTED** [review]
Thanks @albi3ro !

I only spotted a few typos. For some statements, I suggested a re-formulation (using Grammarly). However, I am not an English speaker, so feel free to ignore/change the suggestion if it does not sound right :+1:

**Review by PietropaoloFrisoni — APPROVED** [review]

File: pennylane/tape/qscript.py [hunk 2]
```diff
@@ -889,44 +889,96 @@ def expand(self, depth=1, stop_at=None, expand_measurements=False):
...
-                nested_script,
-                qml.CNOT(wires=[0, 'a']), qml.RY(0.2, wires='a'),
-            ]
-        >>> measurements = [qml.probs(wires=0), qml.probs(wires='a')]
-        >>> qscript = QuantumScript(ops, measurements)
+            This method cannot be used with a tape with non-commuting measurements, even if
+            ``expand_measurements=False``.
 
-        The nested structure is preserved:
+            >>> tape = qml.tape.QuantumScript([], [qml.expval(qml.X(0)), qml.expval(qml.Y(0))])
```
Comment (lillian542) [inline pennylane/tape/qscript.py:900]: This line is too long for the `warning` box, even at full screen width it's pretty narrow. So are the two `QuantumFunctionError` outputs below.

**Review by lillian542 — APPROVED** [review]
Just one suggestion about some lines that are too long for the code blocks - looks good to me!

File: pennylane/tape/qscript.py (hunk 2 above)
Comment (albi3ro) [reply pennylane/tape/qscript.py]: shrinking some line lengths now. thanks for the sugggestion.

**Pull request merged**